packages/web3-core/src/web3_context.ts

```diff
@@ -250,22 +253,85 @@ export class Web3Context<
 		Object.assign(this, _pluginObject);
 	}
 
+	/**
+	 * @returns Returns the current provider
+	 *
+	 * ```ts
+	 * const web3 = new Web3Context("http://localhost:8545");
+	 * console.log(web3.provider);
+	 * > HttpProvider {
+	 * 	clientUrl: 'http://localhost:8545',
+	 * 	httpProviderOptions: undefined
+	 *  }
+	 * ```
+	 */
+
 	public get provider(): SupportedProviders<API> | string | undefined {
 		return this.requestManager.provider;
 	}
 
+	/**
+	 * Will set the current provider.
```

Comment:
Could you please add `@param` description for every parameter in this method and in all other methods that is related to RequestManager and SubscriptionManager. Including the methods that already has comments but does not have any parameter description?
Thanks,